### ISSUE TITLE
JProfiler agent checksum failed

### ISSUE THREAD

**Issue by jakesmolka** [paketo-buildpacks/jprofiler#125]
Currently, our pipeline breaks, when trying to check the checksum of the JProfiler agent.

I suppose this gets solved with https://github.com/paketo-buildpacks/jprofiler/pull/124, but there was also https://github.com/paketo-buildpacks/jprofiler/pull/123 with a different checksum. Personally, I can't reproduce neither on my machine. 

So I'm opening this issue to keep track of the problem, in case there's something off with those checksums.

## Expected Behavior

The agent can be downloaded and utilized by the Buildpack successfully.

## Current Behavior

```
16:14:22  [INFO]     [creator]     ===> DETECTING
16:14:22  [INFO]     [creator]     target distro name/version labels not found, reading /etc/os-release file
16:14:22  [INFO]     [creator]     7 of 27 buildpacks participating
16:14:22  [INFO]     [creator]     paketo-buildpacks/ca-certificates   3.8.6
16:14:22  [INFO]     [creator]     paketo-buildpacks/bellsoft-liberica 11.0.0
16:14:22  [INFO]     [creator]     paketo-buildpacks/syft              2.4.0
16:14:22  [INFO]     [creator]     paketo-buildpacks/executable-jar    6.11.3
16:14:22  [INFO]     [creator]     paketo-buildpacks/dist-zip          5.8.5
16:14:22  [INFO]     [creator]     paketo-buildpacks/spring-boot       5.31.2
16:14:22  [INFO]     [creator]     paketo-buildpacks/jprofiler         7.2.1
16:14:22  [INFO]     [creator]     ===> RESTORING
16:14:22  [INFO]     [creator]     ===> BUILDING
16:14:22  [INFO]     [creator]     target distro name/version labels not found, reading /etc/os-release file
16:14:22  [INFO]     [creator]     
16:14:22  [INFO]     [creator]     Paketo Buildpack for CA Certificates 3.8.6
16:14:22  [INFO]     [creator]       https://github.com/paketo-buildpacks/ca-certificates
16:14:22  [INFO]     [creator]       Build Configuration:
16:14:22  [INFO]     [creator]         $BP_EMBED_CERTS                    false  Embed certificates into the image
16:14:22  [INFO]     [creator]         $BP_ENABLE_RUNTIME_CERT_BINDING    true   Deprecated: Enable/disable certificate helper layer to add certs at runtime
16:14:22  [INFO]     [creator]         $BP_RUNTIME_CERT_BINDING_DISABLED  false  Disable certificate helper layer to add certs at runtime
16:14:22  [INFO]     [creator]       Launch Helper: Contributing to layer
16:14:22  [INFO]     [creator]         Creating /layers/paketo-buildpacks_ca-certificates/helper/exec.d/ca-certificates-helper
16:14:22  [INFO]     [creator]     
16:14:22  [INFO]     [creator]     Paketo Buildpack for BellSoft Liberica 11.0.0
16:14:22  [INFO]     [creator]       https://github.com/paketo-buildpacks/bellsoft-liberica
16:14:22  [INFO]     [creator]       Build Configuration:
16:14:22  [INFO]     [creator]         $BP_JVM_JLINK_ARGS           --no-man-pages --no-header-files --strip-debug --compress=1  configure custom link arguments (--output must be omitted)
16:14:22  [INFO]     [creator]         $BP_JVM_JLINK_ENABLED        false                                                        enables running jlink tool to generate custom JRE
16:14:22  [INFO]     [creator]         $BP_JVM_TYPE                 JRE                                                          the JVM type - JDK or JRE
16:14:22  [INFO]     [creator]         $BP_JVM_VERSION              21                                                           the Java version
16:14:22  [INFO]     [creator]       Launch Configuration:
16:14:22  [INFO]     [creator]         $BPL_DEBUG_ENABLED           false                                                        enables Java remote debugging support
16:14:22  [INFO]     [creator]         $BPL_DEBUG_PORT              8000                                                         configure the remote debugging port
16:14:22  [INFO]     [creator]         $BPL_DEBUG_SUSPEND           false                                                        configure whether to suspend execution until a debugger has attached
16:14:22  [INFO]     [creator]         $BPL_HEAP_DUMP_PATH                                                                       write heap dumps on error to this path
16:14:22  [INFO]     [creator]         $BPL_JAVA_NMT_ENABLED        true                                                         enables Java Native Memory Tracking (NMT)
16:14:22  [INFO]     [creator]         $BPL_JAVA_NMT_LEVEL          summary                                                      configure level of NMT, summary or detail
16:14:22  [INFO]     [creator]         $BPL_JFR_ARGS                                                                             configure custom Java Flight Recording (JFR) arguments
16:14:22  [INFO]     [creator]         $BPL_JFR_ENABLED             false                                                        enables Java Flight Recording (JFR)
16:14:22  [INFO]     [creator]         $BPL_JMX_ENABLED             false                                                        enables Java Management Extensions (JMX)
16:14:22  [INFO]     [creator]         $BPL_JMX_PORT                5000                                                         configure the JMX port
16:14:22  [INFO]     [creator]         $BPL_JVM_HEAD_ROOM           0                                                            the headroom in memory calculation
16:14:22  [INFO]     [creator]         $BPL_JVM_LOADED_CLASS_COUNT  35% of classes                                               the number of loaded classes in memory calculation
16:14:22  [INFO]     [creator]         $BPL_JVM_THREAD_COUNT        250                                                          the number of threads in memory calculation
16:14:22  [INFO]     [creator]         $JAVA_TOOL_OPTIONS                                                                        the JVM launch flags
16:14:22  [INFO]     [creator]         Using Java version 17 extracted from MANIFEST.MF
16:14:22  [INFO]     [creator]       BellSoft Liberica JRE 17.0.13: Contributing to layer
16:14:22  [INFO]     [creator]         Downloading from https://github.com/bell-sw/Liberica/releases/download/17.0.13+12/bellsoft-jre17.0.13+12-linux-amd64.tar.gz
16:14:22  [INFO]     [creator]         Verifying checksum
16:14:22  [INFO]     [creator]         Expanding to /layers/paketo-buildpacks_bellsoft-liberica/jre
16:14:24  [INFO]     [creator]         Adding 146 container CA certificates to JVM truststore
16:14:25  [INFO]     [creator]         Writing env.launch/BPI_APPLICATION_PATH.default
16:14:25  [INFO]     [creator]         Writing env.launch/BPI_JVM_CACERTS.default
16:14:25  [INFO]     [creator]         Writing env.launch/BPI_JVM_CLASS_COUNT.default
16:14:25  [INFO]     [creator]         Writing env.launch/BPI_JVM_SECURITY_PROVIDERS.default
16:14:25  [INFO]     [creator]         Writing env.launch/JAVA_HOME.default
16:14:25  [INFO]     [creator]         Writing env.launch/JAVA_TOOL_OPTIONS.append
16:14:25  [INFO]     [creator]         Writing env.launch/JAVA_TOOL_OPTIONS.delim
16:14:25  [INFO]     [creator]         Writing env.launch/MALLOC_ARENA_MAX.default
16:14:25  [INFO]     [creator]       Launch Helper: Contributing to layer
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/java-opts
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/jvm-heap
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/link-local-dns
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/memory-calculator
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/security-providers-configurer
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/jmx
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/jfr
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/openssl-certificate-loader
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/security-providers-classpath-9
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/debug-9
16:14:25  [INFO]     [creator]         Creating /layers/paketo-buildpacks_bellsoft-liberica/helper/exec.d/nmt
16:14:25  [INFO]     [creator]       Java Security Properties: Contributing to layer
16:14:25  [INFO]     [creator]         Writing env.launch/JAVA_SECURITY_PROPERTIES.default
16:14:25  [INFO]     [creator]         Writing env.launch/JAVA_TOOL_OPTIONS.append
16:14:25  [INFO]     [creator]         Writing env.launch/JAVA_TOOL_OPTIONS.delim
16:14:25  [INFO]     [creator]     
16:14:25  [INFO]     [creator]     Paketo Buildpack for Syft 2.4.0
16:14:25  [INFO]     [creator]       https://github.com/paketo-buildpacks/syft
16:14:25  [INFO]     [creator]         Downloading from https://github.com/anchore/syft/releases/download/v1.15.0/syft_1.15.0_linux_amd64.tar.gz
16:14:25  [INFO]     [creator]         Verifying checksum
16:14:25  [INFO]     [creator]         Writing env.build/SYFT_CHECK_FOR_APP_UPDATE.default
16:14:25  [INFO]     [creator]     
16:14:25  [INFO]     [creator]     Paketo Buildpack for Executable JAR 6.11.3
16:14:25  [INFO]     [creator]       https://github.com/paketo-buildpacks/executable-jar
16:14:44  [INFO]     [creator]       Class Path: Contributing to layer
16:14:44  [INFO]     [creator]         Writing env/CLASSPATH.delim
16:14:44  [INFO]     [creator]         Writing env/CLASSPATH.prepend
16:14:44  [INFO]     [creator]       Process types:
16:14:44  [INFO]     [creator]         executable-jar: java org.springframework.boot.loader.launch.JarLauncher (direct)
16:14:44  [INFO]     [creator]         task:           java org.springframework.boot.loader.launch.JarLauncher (direct)
16:14:44  [INFO]     [creator]         web:            java org.springframework.boot.loader.launch.JarLauncher (direct)
16:14:44  [INFO]     [creator]     
16:14:44  [INFO]     [creator]     Paketo Buildpack for Spring Boot 5.31.2
16:14:44  [INFO]     [creator]       https://github.com/paketo-buildpacks/spring-boot
16:14:44  [INFO]     [creator]       Build Configuration:
16:14:44  [INFO]     [creator]         $BPL_JVM_CDS_ENABLED                 false  whether to enable CDS optimizations at runtime
16:14:44  [INFO]     [creator]         $BPL_SPRING_AOT_ENABLED              false  whether to enable Spring AOT at runtime
16:14:44  [INFO]     [creator]         $BP_JVM_CDS_ENABLED                  false  whether to enable CDS & perform JVM training run
16:14:44  [INFO]     [creator]         $BP_SPRING_AOT_ENABLED               false  whether to enable Spring AOT
16:14:44  [INFO]     [creator]         $BP_SPRING_CLOUD_BINDINGS_DISABLED   false  whether to contribute Spring Boot cloud bindings support
16:14:44  [INFO]     [creator]         $BP_SPRING_CLOUD_BINDINGS_VERSION    1      default version of Spring Cloud Bindings library to contribute
16:14:44  [INFO]     [creator]       Launch Configuration:
16:14:44  [INFO]     [creator]         $BPL_SPRING_CLOUD_BINDINGS_DISABLED  false  whether to auto-configure Spring Boot environment properties from bindings
16:14:44  [INFO]     [creator]         $BPL_SPRING_CLOUD_BINDINGS_ENABLED   true   Deprecated - whether to auto-configure Spring Boot environment properties from bindings
16:14:44  [INFO]     [creator]       Creating slices from layers index
16:14:44  [INFO]     [creator]         dependencies (257.3 MB)
16:14:44  [INFO]     [creator]         spring-boot-loader (409.9 KB)
16:14:44  [INFO]     [creator]         snapshot-dependencies (0.0 B)
16:14:44  [INFO]     [creator]         application (3.6 MB)
16:14:44  [INFO]     [creator]       Spring Cloud Bindings 2.0.3: Contributing to layer
16:14:44  [INFO]     [creator]         Downloading from https://repo1.maven.org/maven2/org/springframework/cloud/spring-cloud-bindings/2.0.3/spring-cloud-bindings-2.0.3.jar
16:14:44  [INFO]     [creator]         Verifying checksum
16:14:44  [INFO]     [creator]         Copying to /layers/paketo-buildpacks_spring-boot/spring-cloud-bindings
16:14:44  [INFO]     [creator]       Web Application Type: Contributing to layer
16:14:44  [INFO]     [creator]         Servlet web application detected
16:14:44  [INFO]     [creator]         Writing env.launch/BPL_JVM_THREAD_COUNT.default
16:14:44  [INFO]     [creator]       Launch Helper: Contributing to layer
16:14:44  [INFO]     [creator]         Creating /layers/paketo-buildpacks_spring-boot/helper/exec.d/spring-cloud-bindings
16:14:44  [INFO]     [creator]       4 application slices
16:14:44  [INFO]     [creator]       Image labels:
16:14:44  [INFO]     [creator]         org.springframework.boot.version
16:14:44  [INFO]     [creator]     
16:14:44  [INFO]     [creator]     Paketo Buildpack for JProfiler 7.2.1
16:14:44  [INFO]     [creator]       https://github.com/paketo-buildpacks/jprofiler
16:14:44  [INFO]     [creator]       JProfiler Agent 14.0.5: Contributing to layer
16:14:44  [INFO]     [creator]         Downloading from https://download-gcdn.ej-technologies.com/jprofiler/jprofiler_linux_14_0_5.tar.gz
16:14:44  [INFO]     [creator]         Verifying checksum
16:14:44  [INFO]     [creator]     unable to invoke layer creator
16:14:44  [INFO]     [creator]     unable to get dependency JProfiler Agent. see DEBUG log level
16:14:44  [INFO]     [creator]     ERROR: failed to build: exit status 1
16:14:45  [INFO] ------------------------------------------------------------------------
16:14:45  [INFO] BUILD FAILURE
16:14:45  [INFO] ------------------------------------------------------------------------
16:14:45  [INFO] Total time:  01:11 min
16:14:45  [INFO] Finished at: 2024-11-06T15:14:45Z
16:14:45  [INFO] ------------------------------------------------------------------------
16:14:45  [ERROR] Failed to execute goal org.springframework.boot:spring-boot-maven-plugin:3.3.5:build-image (default-cli) on project XXXXX: Execution default-cli of goal org.springframework.boot:spring-boot-maven-plugin:3.3.5:build-image failed: Builder lifecycle 'creator' failed with status code 51 -> [Help 1]
16:14:45  [ERROR] 
16:14:45  [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
16:14:45  [ERROR] Re-run Maven using the -X switch to enable full debug logging.
16:14:45  [ERROR] 
16:14:45  [ERROR] For more information about the errors and possible solutions, please read the following articles:
16:14:45  [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

## Possible Solution

Maybe https://github.com/paketo-buildpacks/jprofiler/pull/124?

## Steps to Reproduce

1. Apply the current version of this JProfiler Buildpack

## Motivations

Being able to use this Buildpack.

**Comment by dmikusa** [comment]
Sorry about that. Not sure what happened there. 

We bundled up the new release yesterday and then the checksum on the download and then today the checksum, and only the checksum, from JProfiler changed. This happens sometimes if the company replaces the binary on their download page but doesn't cut a new release. Companies shouldn't do that, it's sketchy, but I have seen it happen.

Anyway, I updated the hash in the buildpack and cut a new buildpack release just now.
